extension/src/popup/helpers/fetch.ts

```diff
@@ -0,0 +1,14 @@
+export const fetchJson = async <T>(url: string, options?: RequestInit) => {
+  const res = await fetch(url, options);
+  if (!res.ok) {
+    throw new Error(res.statusText);
+  }
+
+  if (res.headers.get("content-type") !== "application/json") {
+    const content = await res.text();
+    throw new Error(content);
+  }
+
+  const data = res.json() as T;
```

Comment:
should this be `await res.json()`?

Reply:
hey thanks! fixed in 5be529182068ae3c9a37ae0616fb7e5a16e4998b